### PR TITLE
Local storage/cont

### PR DESCRIPTION
**Pull request by Nathelene**
Did this break anything?
- [ ] No
- [x ]  Yes
Type of change
- [x ] New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.
Checklist:
- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

### CODE REVIEW

**Comment by Nathelene** [comment]
Reviewed and contributed as a group. No errors, everything looks good!

**Comment by BenHolden010** [comment]
The bugs you fixed were very complicated and i though you handled them very well!

File: src/App.tsx [hunk 1]
```diff
@@ -9,37 +9,81 @@ import Nav from "./Components/Nav"
 import {Routes, Route, NavLink} from 'react-router-dom'
 import ServerError from './Components/ServerError';
 import SavedLocations from './Components/SavedLocations';
+import { clear } from 'console';
```
Comment (hfaerber) [inline src/App.tsx:12]: Ooh I'm curious to learn more about this line and why y'all needed it.  I love seeing something in y'alls project code that we didn't explicitly teach you. Nice work!

File: src/App.tsx [hunk 2]
```diff
@@ -9,37 +9,81 @@ import Nav from "./Components/Nav"
...
+    console.log('DATA', data)
+    if (data !== null) setSavedLocations(JSON.parse(data))
+  },[])
+
+
+  useEffect(() => {
+    if(savedLocations) {
+      window.sessionStorage.setItem("SESSION_STORAGE_KEY", JSON.stringify(savedLocations))
+    }
+  }, [savedLocations])
```
Comment (hfaerber) [inline src/App.tsx:58]: love seeing this use of the dependency array

File: src/App.tsx [hunk 3]
```diff
@@ -82,14 +144,11 @@ console.log(locationName, 'NAME')
...
-         locationName={locationName} 
-        //  conditionText={conditionText}
-        //  conditionIcon={conditionIcon} locationRegion={locationRegion} locationCountry={locationCountry} 
-         />}
+        <Route path=":location" element={<FocusCard temp={temp} addLocation={addLocation}
+         locationName={locationName}  locationRegion={locationRegion} conditionText={conditionText}
+         conditionIcon={conditionIcon} locationCountry={locationCountry} />}
          />
-        <Route path='/saved-locations' element={<SavedLocations name={locationName} />} />
+        <Route path='/saved-locations' element={<SavedLocations clearInputs={clearInputs} savedLocations={savedLocations} /> } />
```
Comment (hfaerber) [inline src/App.tsx:151]: You will want to make sure you have a * route for handling bad routes.  You should be able to check that bad routes are handled properly by checking "localhost:3000/nonsense" and "localhost:3000/:somelocation/nonsense".    Users should see messaging like a 404 telling them that page doesn't exist and should have a way to navigate back to the homepage.

**Review by hfaerber — COMMENTED** [review]
really nice work here.  excited to see the final product!

**Pull request merged**